docs/docs/concepts/foundation/communication/public_private_calls/slow_updates_tree.md

```diff
@@ -0,0 +1,97 @@
+---
+title: Privately access Historic Public Data
+---
+
+In Aztec, private and public execution environments are completely separate and operate with distinct state management. It is not possible for private functions to reliably access the most recent public data public state - only sequencers can do that.
+
+But, what about historical public data (or public data that changes infrequently)? Through a **slow updates tree**, private functions can access historical public state. Please note that we are still experimenting with this feature.
+
+On this page you will learn:
+
+1. Why a slow updates tree exists & use cases
+2. How it works
+3. How it can be used to access historical public data
+4. Limitations
+
+## The need for a slow updates tree
+
+This structure was created specifically to privately & publicly access historical public data that is not updated often. It should be used to store data that is not sensitive but needs to be accessible from private smart contracts. Using a slow updates tree, it is possible to:
+
+- Access historic public data from a private function
+- Access historic public data from a public function
+- Update public data (that does not need updated often) from public and private functions 
+
+This data structure is ideal for these use cases:
+
+- Address Registry: Enabling contracts to interact with other contracts more easily requires address storage accessible in both public and private executions. This can be particularly helpful in things such as proxy contracts.
+- Access Control: Managing privileges in contracts, such as a token contract owner’s ability to mint new tokens, is streamlined when control information is shared between public and private executions. This might include things like blacklists and whitelists.
+
+## How it works
+
+We developed the Slow Update Tree to help balance public and private execution in a blockchain context.
+
+Earlier systems typically used either fully public or entirely private state trees, which led to privacy leaks if reading public data from private state.
+
+Using a shared state tree, it is possible to privately access historic public data by providing a membership proof to show that a value is indeed part of a commitment, and then check that the commitment matches the one stored in the state, as shown below.
+
+```mermaid
+graph TD;
+    Cm[CommitmentToKnowingAValue] --> Vm1[Value1]
+    Cm --> Vm2[Value2]
+    Cm --> Vmn[Value3]
+    Vm1 --> Mp1[Membership Proof for Value1]
+    Mp1 --> St[Commitment in State]
+    St --> Cm
+```
+
+However this means that any changes to the commitment will invalidate our membership proof, and make our read fail.
+
+To solve this, the Slow Update Tree is a dual-tree structure - using a current and pending tree - that updates at the end of an epoch. This way, we can ensure that the values are stable throughout the epoch, and that the membership proofs are not invalidated by changes in other contracts more than once every epoch.
```

Comment:
to prevent confusion down the line, can you explain the use of the pending tree -> that any updates that need to be made go into the pending tree and after an epoch, they become the current tree. Anf then say, by having two separate trees, we can ensure the valyes are stable...